src/node_contextify.h

```diff
@@ -98,6 +102,47 @@ class ContextifyContext {
       const v8::PropertyCallbackInfo<v8::Boolean>& args);
 };
 
+class ContextifyScript : public BaseObject {
+ private:
+  Persistent<v8::UnboundScript> script_;
+
+  static void New(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static bool InstanceOf(Environment* env, const v8::Local<v8::Value>& value);
+  static void RunInThisContext(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void RunInContext(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void DecorateErrorStack(
+      Environment* env,
+      const v8::TryCatch& try_catch);
+  static bool EvalMachine(
+      Environment* env,
+      const int64_t timeout,
+      const bool display_errors,
+      const bool break_on_sigint,
+      const v8::FunctionCallbackInfo<v8::Value>& args);
+
+
+  static int Identity_;
+
+ public:
+  static void Init(Environment* env, v8::Local<v8::Object> target);
+  static ContextifyScript* GetFromID(v8::Local<v8::Context>, int id);
+  int id;
+
+  ContextifyScript(Environment* env, v8::Local<v8::Object> object)
+    : BaseObject(env, object) {
+      MakeWeak<ContextifyScript>(this);
+      id = ContextifyScript::Identity_++;
+  }
+
+  ~ContextifyScript() {
+    // important to use GetCurrentContext so that we get the right dataset
+    auto data = GetModuleData(env()->isolate()->GetCurrentContext());
```

Comment:
This won't work if there are no JavaScript stack frames on top, which can happen during GC. (Same applies to `ModuleWrap`.)

This makes me think that everything in `ContextModuleData` should in fact live in `IsolateData`.